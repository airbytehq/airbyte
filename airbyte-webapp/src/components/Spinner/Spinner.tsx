import React from "react";
import styled, { keyframes } from "styled-components";

interface IProps {
  backgroundColor?: string;
  small?: boolean;
}

export const SpinAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const SpinnerWheel = styled.div<{ small?: boolean }>`
  display: inline-block;
  height: ${({ small }) => (small ? 30 : 42)}px;
  width: ${({ small }) => (small ? 30 : 42)}px;
  border-radius: 50%;
  border: 4px solid ${({ theme }) => theme.primaryColor12};
  position: relative;
  animation: ${SpinAnimation} 1.5s linear 0s infinite;
`;

const BreakRec = styled.div<IProps>`
  width: 13px;
  height: 7px;
  background: ${({ theme, backgroundColor }) => (backgroundColor ? backgroundColor : theme.whiteColor)};
  top: -4px;
  position: relative;
  margin: 0 auto;
`;

const Spinner: React.FC<IProps> = ({ backgroundColor, small }) => (
  <SpinnerWheel small={small}>
    <BreakRec backgroundColor={backgroundColor} />
  </SpinnerWheel>
);

export default Spinner;
