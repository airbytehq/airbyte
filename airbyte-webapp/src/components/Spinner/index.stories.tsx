import { ComponentStory, ComponentMeta } from "@storybook/react";

import SpinnerComponent from "./Spinner";

export default {
  title: "Ui/Spinner",
  component: SpinnerComponent,
  argTypes: {
    small: { type: "boolean", required: false },
  },
} as ComponentMeta<typeof SpinnerComponent>;

const Template: ComponentStory<typeof SpinnerComponent> = (args) => <SpinnerComponent {...args} />;

export const Primary = Template.bind({});
