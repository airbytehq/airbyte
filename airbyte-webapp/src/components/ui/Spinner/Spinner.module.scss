@use "../../../scss/colors";

@keyframes spin {
  from {
    transform: rotate(0);
  }

  to {
    transform: rotate(359deg);
  }
}

.spinner {
  width: 42px;
  height: 42px;
  border: 4px solid colors.$blue-100;
  border-top: 4px solid transparent;
  border-radius: 50%;
  animation: spin 1.5s linear 0s infinite;
}

.small {
  width: 30px;
  height: 30px;
}
