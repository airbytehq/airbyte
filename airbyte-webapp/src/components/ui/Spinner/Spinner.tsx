import classnames from "classnames";
import React from "react";

import styles from "./Spinner.module.scss";

interface SpinnerProps {
  small?: boolean;
}

export const Spinner: React.FC<SpinnerProps> = ({ small }) => (
  <div className={classnames(styles.spinner, { [styles.small]: small })} />
);
