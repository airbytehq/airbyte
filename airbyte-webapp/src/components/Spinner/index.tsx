import Spinner from "./Spinner";

export default Spinner;
