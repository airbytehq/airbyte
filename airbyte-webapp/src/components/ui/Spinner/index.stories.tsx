import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Spinner } from "./Spinner";

export default {
  title: "UI/Spinner",
  component: Spinner,
  argTypes: {
    small: { type: "boolean", required: false },
  },
} as ComponentMeta<typeof Spinner>;

const Template: ComponentStory<typeof Spinner> = (args) => <Spinner {...args} />;

export const Primary = Template.bind({});
