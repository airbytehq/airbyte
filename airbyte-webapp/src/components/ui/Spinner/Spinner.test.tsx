import { render } from "@testing-library/react";

import { Spinner } from "./Spinner";

describe("<Spinner />", () => {
  it("should render without crash", () => {
    const component = render(<Spinner />);

    expect(component).toMatchSnapshot();
  });
});
