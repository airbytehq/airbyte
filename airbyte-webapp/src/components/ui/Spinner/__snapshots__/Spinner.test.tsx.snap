// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render without crash 1`] = `
<body>
  <div>
    <div
      class="<removed-for-snapshot-test>"
    />
  </div>
</body>
`;
